File: week_1/day_1_arrays/move_zero_follow_up_questions/What_if_zero_is_replaced_with_a_value_X.py
def moveValueToEnd(nums, X):
    j = 0
    for i in range(len(nums)):
        if nums[i] != X:
            nums[i], nums[j] = nums[j], nums[i]
            j += 1

nums = [2, 1, 2, 3, 2, 4, 2]
moveValueToEnd(nums, 2)
print(nums)

File: week_1/day_1_arrays/move_zero_follow_up_questions/What_if_zero_is_replaced_with_a_value_X.js
function moveValueToEnd(nums, X) {
    let j = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] !== X) {
            [nums[i], nums[j]] = [nums[j], nums[i]];
            j++;
        }
    }
}

let nums = [2, 1, 2, 3, 2, 4, 2];
moveValueToEnd(nums, 2);
console.log(nums);
